hessian/src/main/java/com/linecorp/armeria/internal/server/hessian/HessianHttpServiceImpl.java

```diff
@@ -0,0 +1,579 @@
+                                                 serializerFactory, cause);
+        respond(serializationFormat, content, httpRes);
+    }
+
+    private static void respond(SerializationFormat serializationFormat, HttpData content,
+                                CompletableFuture<HttpResponse> res) {
+        res.complete(HttpResponse.of(HttpStatus.OK, serializationFormat.mediaType(), content));
+    }
+
+    private static HttpData encodeSuccess(ServiceRequestContext ctx,
+                                          HeaderType headerType,
+                                          SerializerFactory serializerFactory,
+                                          Object value) {
+
+        final ByteBuf buf = ctx.alloc().buffer(128);
+        boolean success = false;
+        final AbstractHessianOutput hessianOutput;
+        try {
+            hessianOutput = hessianOutput(buf, headerType, serializerFactory);
+            hessianOutput.writeReply(value);
+            // not need to close.
+            hessianOutput.flush();
+            final HttpData encoded = HttpData.wrap(buf);
+            success = true;
+            return encoded;
+        } catch (IOException e) { // NOSONAR
+            logger.error("Write hessian response error.", e);
+            throw new IllegalStateException("Write hessian response error", e);
+        } finally {
+            if (!success) {
+                buf.release();
+            }
+        }
+    }
+
+    private static HttpData encodeException(ServiceRequestContext ctx,
+                                            HeaderType headerType,
+                                            SerializerFactory serializerFactory,
+                                            Throwable cause) {
+
+        Throwable exception = cause;
+        if (exception instanceof InvocationTargetException) {
+            exception = ((InvocationTargetException) cause).getTargetException();
+        }
+
+        logger.info("encode exception.", exception);
+
+        final ByteBuf buf = ctx.alloc().buffer(128);
+        boolean success = false;
+
+        final AbstractHessianOutput out;
+        try {
+            out = hessianOutput(buf, headerType, serializerFactory);
+            if (exception instanceof HessianFaultException) {
+                final HessianFaultException hfe = (HessianFaultException) exception;
+                out.writeFault(hfe.getCode(), escapeMessage(hfe.getMessage()), hfe.getDetail());
+            } else {
+                out.writeFault("ServiceException", escapeMessage(exception.getMessage()), exception);
+            }
+            // flush, not close.
+            out.flush();
+            final HttpData encoded = HttpData.wrap(buf);
+            success = true;
+            return encoded;
+        } catch (IOException e) { // NOSONAR
+            logger.error("write hessian error response failed:", e);
+            throw new IllegalStateException("Write error response failed.", e);
```

Comment:
No need to log if you're propagating an exception, because otherwise it will be double notification.
```suggestion
            throw new IllegalStateException("Failed to encode an error response:", e);
```